portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,28 +14,107 @@
 
 package com.google.sps.servlets;
 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.FetchOptions.Builder;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
 import com.google.gson.Gson;
+import com.google.sps.data.Comment;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import java.util.ArrayList; 
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
-@WebServlet("/data")
+/** Servlet responsible for listing and adding comments data */
+@WebServlet("/comment")
 public class DataServlet extends HttpServlet {
 
+  private enum Sort { NEW, OLD, NULL }
+  
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    ArrayList<String> messages = new ArrayList<String>();
-    messages.add("Hello");
-    messages.add("Testing JSON");
-    messages.add("I hope this works!");
+    Sort sort = Sort.valueOf(request.getParameter("sort").toUpperCase());
+    String imgRequest = request.getParameter("img");
+
+    // Sort comments according to request parameter
+    Query query;
+    if (sort == Sort.NULL || sort == Sort.NEW) {
+      query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    } else {
+      query = new Query("Comment").addSort("timestamp", SortDirection.ASCENDING);
```

Comment:
No need to change but what i would do for 

SortDirection sortDirection = (Sort.NULL || sort == Sort.NEW) ?  SortDirection.DESCENDING : SortDirection.ASCENDING;
Query query =  new Query("Comment").addSort("timestamp", sortDirection);